contracts/cosoul/SBT.sol

```diff
@@ -0,0 +1,216 @@
+		require(_slot < 8);
+		uint256 value = getSlot(_slot, _tokenId);
+		require(value + _amount <= type(uint32).max, "CoSoul: uint32 overflow");
+		uint256 current = blobs[_tokenId];
+		blobs[_tokenId] = current + (_amount << _slot);
+	}
+
+	/**
+	 * @notice
+	 * Function to decrement the value of a slot in a blob by some amount
+	 * @param _slot Slot value. Up to 7
+	 * @param _amount Amout to decrement a slot
+	 * @param _tokenId Token ID from which to update the blob data
+	 */
+	function decSlot(uint8 _slot, uint256 _amount, uint256 _tokenId) external authorised(msg.sender) {
+		require(_slot < 8);
+		uint256 value = getSlot(_slot, _tokenId);
+		require(value >= _amount, "CoSoul: uint32 overflow");
+		uint256 current = blobs[_tokenId];
+		blobs[_tokenId] = current - (_amount << _slot);
+	}
+
+	/**
+	 * @notice
+	 * Function to sync blob data of a token from a merkle tree signed by our signer
+	 * @param _data Blob data that will overwrite current data
+	 * @param _tokenId Token ID from which to update blob
+	 * @param _nonce Sync counter used to prevent replays
+	 * @param _signature Signature provided by our signer to validate leaf data
+	 */
+	function syncWithSignature(uint256 _data ,uint256 _tokenId, uint256 _nonce, bytes calldata _signature) external {
+		require(ownerOf(_tokenId) == msg.sender);
+		require(syncNonces[_nonce]++ == _nonce);
+		require(keccak256(abi.encodePacked(_tokenId, _nonce, _data)).toEthSignedMessageHash().recover(_signature) == signer, "Sig not valid");
+
+		blobs[_tokenId] = _data;
+	}
+
+	/**
+	 * @notice
+	 * Function to transfer token under approval of the protocol. Gated by authorised addresses
+	 * @param _from Previous token owner
+	 * @param _to New token owner
+	 * @param _tokenId Token to transfer
+	 */
+	function overrideTransfer(address from, address _to, uint256 _tokenId) external authorised(msg.sender) {
+		_transfer(_from, _to, _tokenId);
+	}
+
+	/**
+	 * @notice
+	 * Function to transfer token under approval of the protocol via signature
+	 * @param _from Previous token owner
+	 * @param _to New token owner
+	 * @param _tokenId Token to transfer
+	 * @param _nonce Transfer counter used to prevent replays
+	 * @param _signature Signature provided by our signer to validate transfer
+	 */
+	function overrideTransferWithSignature(
+		address _from,
+		address _to,
+		uint256 _tokenId,
+		uint256 _nonce,
+		bytes calldata _signature) external {
+		require(ownerOf(_tokenId) == msg.sender);
+		require(transferNonces[_tokenId]++ == _nonce);
+		require(keccak256(abi.encodePacked(_tokenId, _nonce)).toEthSignedMessageHash().recover(_signature) == signer, "Sig not valid");
+
+		_transfer(_from, _to, _tokenId);
+	}
+
+	/**
+	 * @notice
+	 * Function to mint token via signature to msg.sender
+	 * @param _nonce Mint counter used to prevent replays
+	 * @param _signature Signature provided by our signer to validate mint
+	 */
+	function mintWithSignature(uint256 _nonce, bytes calldata _signature) external {
```

Comment:
Can you explain this mintWithSignature, vs just a mint function that doesn't require a signature?

Reply:
I'm assuming this pattern: https://portal.thirdweb.com/glossary/signature-based-minting